python/pyspark/sql/streaming/listener.py

```diff
@@ -87,6 +91,14 @@ def onQueryProgress(self, event: "QueryProgressEvent") -> None:
         """
         pass
 
+    @abstractmethod
+    def onQueryIdle(self, event: "QueryIdleEvent") -> None:
+        """
+        Called when the query is idle for a certain time period and waiting for new data to
```

Comment:
`a certain time period` sounds ambiguous.

Reply:
Yeah maybe better to explicitly mention about config. Shouldn't be risky as config is another form of public API which we don't change. Will do.

Reply:
Sigh the config is internal, not public. I'll just shorten as idle which implicitly includes the semantic "for some amount of time". I'll also change the doc in the config so that it reflects to the new behavior. Please let me know if we think it's better to make the config be public instead.

Reply:
Doc change looks okay to me.